### ISSUE TITLE
"Override Method" functionality issues

### ISSUE THREAD

**Issue by kitlbast** [sadovnychyi/autocomplete-python#186]
The first invoke of override method doesn't show an option with Parent.__init__ method. This method present in options only after choosing object.__init__ for the first time. Also, self.__class__ inserted instead of actual class name.

![override_issue_record2](https://cloud.githubusercontent.com/assets/59706/13527057/0a206f28-e215-11e5-88db-faca6944767d.gif)

**Comment by sadovnychyi** [comment]
Yeah, it can be buggy due to the hacky implementation. I'll take a look on your specific case once I have some time, but it could be weeks after now.
